### PR TITLE
Allow enabling CORS while in a docker container

### PR DESCRIPTION
**Pull request by gergof**
## Proposed changes

Added an environment variable named CORS_ORIGINS_REGEX that can be used to set up a regex for which the cors requests would be allowed.

## Checklist

- [x] Lint and unit tests pass locally with my changes.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] I have added documentation to describe my feature.
- [x] I have squashed my commits into logic units.
- [x] I have described the changes in the commit messages.

### CODE REVIEW

**Comment by nijel** [comment]
Wouldn't it be more reasonable to set this for `/api/` only? See also https://github.com/WeblateOrg/weblate/issues/3028 (where use case question was not really answered).

Can you please add documentation describing this to https://github.com/WeblateOrg/weblate/blob/main/docs/admin/install/docker.rst?

**Comment by gergof** [comment]
My use case is this:

We have a feature that allows translators to have a live preview of their work directly in the application.
This uses the weblate API to fetch the JSON file and include that one instead of the ones deployed in production. So if they make a change, they can just hit refresh in the application and validate their changes.

**Comment by nijel** [comment]
Okay, so limiting this to /api/ would work for you. Can you please adjust the patch that way?

File: start [hunk 1]
```diff
@@ -243,8 +243,20 @@ set_real_ip_from 0.0.0.0/0;
             ;;
     esac
 
+    if [ -n "${CORS_ORIGINS_REGEX}" ]; then
+        WEBLATE_ENABLE_CORS="
+if (\$http_origin ~ '$CORS_ORIGINS_REGEX') {
+    add_header 'Access-Control-Allow-Origin' "\$http_origin" always;
```
Comment (nijel) [inline start:249]: The quote inside quote seems wrong, either remove it, or add escaping:

```suggestion
    add_header 'Access-Control-Allow-Origin' \"\$http_origin\" always;
```

**Comment by nijel** [comment]
Thanks for the PR, but we're going to integrate https://pypi.org/project/django-cors-headers/ instead.

**Pull request closed without merging**